docs/customresourcestate-metrics.md

```diff
@@ -0,0 +1,220 @@
+# Custom Resource State Metrics
+
+This section describes how to add metrics based on the state of a custom resource without writing a custom resource 
+registry and running your own build of KSM.
+
+## Configuration
+
+The `KSM_CUSTOM_RESOURCE_METRICS` environment variable can be used to configure the metrics for custom resources:
```

Comment:
As environment variables are limited in size to 32k chars, have you considered using a file-based config instead?

Reply:
Oh sure easy as, just prototyping with something simple.